src/Peachpie.Library/OpenSSL.cs

```diff
@@ -0,0 +1,741 @@
+                    }
+                    else // Load from string
+                        return new X509Resource(new X509Certificate2(ctx.StringEncoding.GetBytes(cert)));
+                }
+                catch (CryptographicException)
+                {
+                    PhpException.Throw(PhpError.Warning, Resources.Resources.X509_cannot_be_coerced);
+                    return null;
+                }
+            }
+
+            //
+            PhpException.Throw(PhpError.Warning, Resources.Resources.X509_cannot_be_coerced);
+            return null;
+        }
+
+        /// <summary>
+        /// Parses the certificate supplied by x509certdata and returns a resource identifier for it.
+        /// </summary>
+        /// <param name="ctx">Context of the script.</param>
+        /// <param name="x509certdata">Path to file with PEM encoded certificate or a string containing the content of a certificate or X509Resource</param>
+        /// <returns>Returns a resource identifier on success or FALSE on failure.</returns>
+        [return: CastToFalse]
+        public static X509Resource openssl_x509_read(Context ctx, PhpValue x509certdata)
+        {
+            return ParseX509Certificate(ctx, x509certdata);
+        }
+
+        /// <summary>
+        /// Stores x509 into a file named by outfilename in a PEM encoded format.
+        /// </summary>
+        /// <param name="ctx">Context of the script.</param>
+        /// <param name="x509">Path to file with PEM encoded certificate or a string containing the content of a certificate or X509Resource</param>
+        /// <param name="outfilename">Path to the output file.</param>
+        /// <param name="notext">The optional parameter notext affects the verbosity of the output; if it is FALSE, then additional human-readable information is included in the output. The default value of notext is TRUE.</param>
+        /// <returns>Returns TRUE on success or FALSE on failure.</returns>
+        public static bool openssl_x509_export_to_file(Context ctx, PhpValue x509, string outfilename, bool notext = true)
+        {
+            var resource = ParseX509Certificate(ctx, x509);
+            if (resource == null)
+                return false;
+
+            string certificate = Export(resource, notext);
+
+            try
+            {
+                using (StreamWriter wr = new StreamWriter(outfilename))
+                    wr.Write(certificate);
+            }
+            catch (IOException) {  return false; }
+
+            return true;
+        }
+
+        private static string Export(X509Resource x509, bool notext)
+        {
+            StringBuilder builder = new StringBuilder();
+
+            if (!notext)
+            {
+                builder.AppendLine("Certificate:");
+                builder.AppendLine("\tData:");
+                builder.AppendLine(String.Format("\t\tVersion: {0} 0x{0:X}",x509.Certificate.Version));
```

Comment:
`StringBuilder` already has an overload for string formatting, please use that one. `String.Format` creates another StringBuilder instance inside unnecessarily.